components/identifier-generator/back/src/Infrastructure/Subscriber/SetIdentifiersSubscriber.php

```
@@ -127,4 +152,41 @@ private function getIdentifierGenerators(): array
 
         return $this->identifierGenerators;
     }
+
+    /**
+     * Returns Symfony constraints defined here:
+     * src/Akeneo/Pim/Enrichment/Bundle/Resources/config/validation/product.yml
+     *
+     * @return Constraint[]
+     */
+    private function getProductConstraints(ProductInterface $product): array
+    {
+        $metadata = $this->metadataFactory->getMetadataFor($product);
+        Assert::isInstanceOf($metadata, ClassMetadataInterface::class);
```

that might lead to uncaught exception, this preventing to save of the product. Maybe add a general try/catch somewhere in the `setGeneratorIdentifier` method and cast it to `UnableToSetIdentifierException` ?

---

Don't agree with you ; this is completely internal to Symfony. A MetadataFactory from a class HAS TO return a ClassMetadataInterface. If it's not the case, I prefer to fail with a 500 and a datadog error and failing tests instead of failing silently and not know the root cause.